tests/functions/test_openai.py

```diff
@@ -399,6 +411,116 @@ def test_image_generations_dall_e_3(self, reset_db) -> None:
+        weather_agent.add_computed_column(
+            initial_response=chat_completions(
+                model='gpt-4o-mini',
+                messages=[{'role': 'user', 'content': weather_agent.prompt}],
+                tools=weather_tools,
+                tool_choice=weather_tools.choice(required=True),
+            )
+        )
+        weather_agent.add_computed_column(tool_output=invoke_tools(weather_tools, weather_agent.initial_response))
+        weather_agent.add_computed_column(
+            tool_response_prompt=pxtf.string.format(
+                'Orginal Prompt\n{0}: Tool Output\n{1}', weather_agent.prompt, weather_agent.tool_output
+            )
+        )
+        weather_agent.add_computed_column(
+            final_response=chat_completions(
+                model='gpt-4o-mini',
+                messages=[
+                    {
+                        'role': 'system',
+                        'content': 'You are a helpful AI assistant that can use various tools. Analyze the tool results and provide a clear, concise response.',
+                    },
+                    {'role': 'user', 'content': weather_agent.tool_response_prompt},
+                ],
+            )
+        )
+        weather_agent.add_computed_column(answer=weather_agent.final_response.choices[0].message.content)
+        weather_agent_udf = pxt.udf(weather_agent, return_value=weather_agent.answer)
+
+        # Team tools
+        team_tools = pxt.tools(finance_agent_udf, weather_agent_udf)
+
+        # Manager Agent
+        manager = pxt.create_table('manager', {'prompt': pxt.String})
+        manager.add_computed_column(
+            initial_response=chat_completions(
+                model='gpt-4o-mini',
+                messages=[{'role': 'user', 'content': manager.prompt}],
+                tools=team_tools,
+                tool_choice=team_tools.choice(required=True),
+            )
+        )
+        manager.add_computed_column(tool_output=invoke_tools(team_tools, manager.initial_response))
+        manager.add_computed_column(
+            tool_response_prompt=pxtf.string.format(
+                'Orginal Prompt\n{0}: Tool Output\n{1}', manager.prompt, manager.tool_output
+            )
+        )
+        manager.add_computed_column(
+            final_response=chat_completions(
+                model='gpt-4o-mini',
+                messages=[
+                    {
+                        'role': 'system',
+                        'content': 'You are a helpful AI assistant that can use various tools. Analyze the tool results and provide a clear, concise response.',
+                    },
+                    {'role': 'user', 'content': manager.tool_response_prompt},
+                ],
+            )
+        )
+        manager.add_computed_column(answer=manager.final_response.choices[0].message.content)
+
+        manager.insert([{'prompt': "what's the weather in sf"}])
+        _ = manager.select(manager.answer).collect()
```

Comment:
Best to check the responses here for end-to-end validation that the tools are getting invoked properly.

Reply:
Done.